indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamDataSourceMetadata.java

```diff
@@ -59,6 +59,19 @@ public boolean matches(DataSourceMetadata other)
     return plus(other).equals(other.plus(this));
   }
 
+  @Override
+  public boolean isGreater(DataSourceMetadata other)
```

Comment:
how about implementing the `Comparable` interface for this instead of rolling our own?